pilot/pkg/model/xds_cache.go

```diff
@@ -47,13 +48,20 @@ var (
 		"Total number of xds cache evictions.",
 	)
 
+	xdsCacheErrors = monitoring.NewSum(
+		"xds_cache_errors",
+		"Total number of invalid xds cache access.",
+	)
+
 	xdsCacheSize = monitoring.NewGauge(
 		"xds_cache_size",
 		"Current size of xds cache",
 	)
 
-	xdsCacheHits   = xdsCacheReads.With(typeTag.Value("hit"))
-	xdsCacheMisses = xdsCacheReads.With(typeTag.Value("miss"))
+	xdsCacheHits          = xdsCacheReads.With(typeTag.Value("hit"))
+	xdsCacheMisses        = xdsCacheReads.With(typeTag.Value("miss"))
+	xdsCacheStaleUpdate   = xdsCacheErrors.With(typeTag.Value("stale"))
```

Comment:
stale is not an error, it feels misleading under xds_cache_errors
. I expect this number to be quite high in standard scenario. Anytime we get a config change all in-flight changes will become stale

Reply:
Yeah. Was wondering about that too and it may not be valuable to have this metrics. I will remove it.